README.md
eamod
=====
Now eAmod works under GNU General Public License, and
everybody can read about it at http://www.gnu.org/licenses/gpl.html

eAmod is an extended project of eAthena, a Ragnarok Private Server Emulator, created by
Andres Garbanzo Mora, also known as Zephyrus, owner and administrator of eAthena Modifications

This document contains the list of all features on eAmod-A. I will keep it updated and could be usefull
is somebody don't know about something.


- Battleground Capture the Flag with minimap marks about positions of flags.
- Battleground Eye of the Storm based on World of Warcraft. This combine Domination and Capture the flag systems.
- Battleground TeamDeathMatch.
- Battleground Tierra Boosnia (Alterac Valley Style).
- Battleground Stone Control (Multiple Capture the Flag)
- Battleground Tierra Tiple Infierno (Triple TeamDeathMatch)
- Battleground Conquest (WoE Second Edition Simulation)
- Battleground Supply items shop using badges.
- WoE Supply shop using badges.
- Duel Arena 1 vs 1 with global announces, using the KvM system.
- Expert Battleground Ranking system.

- War of Emperium expert ranking. Take a look where www.chronos-ro.com on Rankings, in Guild, WoE and Castle submenus.
- Multiple War of Emperium events, with different castles each one and autoblocking the non playable castles.
- Ancient WoE, to play with 1st and 2nd class characters only, and equipment and items from 9 season and previous.

- House for Rent System.
- Mining using the WoW system. Mines spawns in specified maps, and different kinds of mines.
- Endless Tower official script.
- Special High Rate NPC all services.
- Game Rooms with Emperium Breaker, Barricade and Guardian Stone simulations, Memory, Audio Memory and Simon Says games.
- Hunting System. Where you receive a list of mobs to kill each 24 hours.
- Cash Shop system, to inspire donations. Premium account system.

- World of Warcraft Channel System.
- Item serialization system, where each Equipment have an unique serial number, so you can easily track items or detect duplicates.
- @security command for players. This allow to set a password to block item drops, trades, any kind of item out.
- @aura to set different auras between players.
- GM Tools scripts to set Delayed Shutdown, or a Player variable editor (to edit or check player quest variables).
- GM Commands for Events.
- @pvpmode to enter a PK mode where you can kill others players anywhere.


- Renewal mechanics. Most of the new formulas (ASPD, Damage, Cards, Defense, etc).
- All 3-1 Classes from 3CeAM, improved and fixed.
- Currently working on 3-2 Classes, one by one.
- Renewal databases: Experience, Monsters (up to Dicastes), Stats Points, ASPD, Items, Elements, etc.
- Renewal Mechanics can be set on/off by just using a battle configuration setting.
- El Dicastes, Scarabah Dungeon, Nyd.


CPanel/CHANGELOG.txt
Date

2008/05/17
	* Changed password.php and changemail.php to POST method. [BeoWulf]
	* Fixed the mail sending function, now the recover password function should work, also
	now that we're using the phpmailer class you should be able to use gmail smtp server too [Dekamaster]
	* Disabled error report by default. [Dekamaster]
	* Added a definition file for the e-mail template, now you can customize the template of
	the email that will be sent to your players in /language/mail/ [Dekamaster]
	* Added Castle names, kisuka report on eAthena forums. [BeoWulf]
	* Added extra protection to login.php after some security leak rumors (not yet confirmed though). [BeoWulf]
	* Testing a new Captcha, its uglier than the original, but more safe. [Dekamaster]
	* Extended some of login.php protection to account.php and recover.php just in case. [BeoWulf]

2008/05/14
	* Fixed a bug with reset position leting a jailed person to get out, reported by panchox
	at ea boards (and thanks for the temporary fix, brianlew, we really appreciated it.) [Dekamaster]

------------------
We are retaking the project today (2008/05/14), we'll be slowly adding new stuffs to the control
panel and fixing old bugs, if you have a bug to report you are welcome to share it with us at
cerescp@gmail.com (yes we are reading it again)
------------------

2007/01/01
	* Fixed Firefox incompatibility, stupid browser. [BeoWulf]

2006/12/03
	* Reviewed img.php and middle.php files with help of gunawan wibisono. [BeoWulf]

2006/10/23
	* Added a variable in recover.php. [BeoWulf]

2006/10/18
	* Fixed an exploit in language.php. [BeoWulf]

2006/10/15
	* Added a one day ipban to users blocked, it will keep then out to think about
	what they did and can't creat accounts with ofensive nicks against GMs. [BeoWulf]
	* Small fix in server_status.php, and a change in account.php so now its
	post not get (internal stuff). [BeoWulf]

2006/10/03
	* Fixed errors in emblems.php, adminquery.php, ladder.php and functions.php. [BeoWulf]

2006/10/03
	* Changed all session maybe now there is no conflict with 2 cps in same machine. [BeoWulf]

2006/09/22
	* Added some fuzzy to code image. [BeoWulf]

2006/09/18
	* Created first trial of an install script. [BeoWulf]
	* Fixed install script checks and w3c instructions. [BeoWulf]
	* Changed INSTALL.TXT instructions to fit the new install script. [BeoWulf]

2006/09/18
	* Added jail check to position.php so jailed characters may not reset
		their position. [Entwined]
	* Added Post functions do ajax engine. [BeoWulf]
	* Changed login.php from get to post. [BeoWulf]
	* Fixed whoisonline.php adjust to show position to minimum admin level set 
		in config.php. [BeoWulf]
	* Added check in functions.php to destroy session if you use more than one
		CP in same machine when changing beteween them use different names 
		for the CPs or it can be exploitable, still working on a different 
		solution. [BeoWulf]
	* Changed admincharinfo.php to a beta test in changing stored char_id in 
		equip slots into something we can read. [BeoWulf]
	* Added adminchars.php still working on it, to manage chars. [BeoWulf]
	* Added money.php testing a 2 minutes ban after transfer. [BeoWulf]
	* Added marriage.php testing a 2 minutes ban after divorce. [BeoWulf]
	* Added guild.php if castle name is not defined it will skip and not show. [BeoWulf]
	* Added dutch and indonesian to languages. [BeoWulf]
	* Fixed some stuff in other language files, but still need translations. [BeoWulf]
	* Fixed ladder.php and about.php to show the new classes. [BeoWulf]

2006/09/14
	* Added map and gm level check for position reset, so you can't reset your position if your in jail [Entwined]
	* Added install/sql/rag_ragsrvinfo.sql and install/npc/agit_status.txt. You will have to install both of 
		these correctly for agit status	check to work. [Entwined]
	* Added ceres.php for control panel information [Entwined]
	* Added max account check for account registration [Entwined]

2006/09/13
	* Added IP address field insertion on registrations (last_ip) [Entwined]
	* Added an email verification field for account registration. [Entwined]
	* Added some config settings for future use. [Entwined]
	* Created the initial repository of CeresCP (rev4) as the working base. [Entwined]


CP/guild.php
<?php
	session_start();
	include_once 'config.php';
	include_once 'functions.php';

	$Rank = array(
		"Civilian",
		"Private",
		"Corporal",
		"Sergeant",
		"Master<br>Sergeant",
		"Sergeant<br>Major",
		"Knight",
		"Knight<br>Lieutenant",
		"Knight<br>Captain",
		"Knight<br>Champion",
		"Lieutenant<br>Commander",
		"Commander",
		"Marshal",
		"Field Marshal",
		"Grand Marshal",
	);

	function calc_rank($score)
	{
		$result = intval($score / 270);
		if( $result > 14 )
			$result = 14;

		return $result;
	}

	if( !isset($_GET['opt']) )
	{
		echo '
			<div class="ladder_class">
		';

		opentable('Ranking General de Guilds');

		echo '
			<table width="600">
				<tr>
					<td align="center">
						<form id="ladder">
							<select name="opt" onChange="javascript:GET_ajax(\'guild.php\', \'ladder_div\', \'ladder\');">
								<option selected value="0">Nivel y Experiencia</option>
								<option value="1">Propiedad de Castillos</option>
								<option value="2">World Conquest War</option>
							</select>
						</form>
					</td>
				</tr>
			</table>
			<div id="ladder_div" style="color:#000000">
		';

		$begin = 1;
		$_GET['opt'] = 0;
	}

	if( $_GET['opt'] == 0 )
	{ // Nivel y Experiencia
?>
	<table width="600">
		<tr>
			<td align="left" class="head">#</td>
			<td align="center" class="head">Emblema</td>
			<td align="left" class="head">Nombre</td>
			<td align="left" class="head">Nivel</td>
			<td align="right" class="head">Experiencia</td>
			<td align="right" class="head">Promedio</td>
		</tr>
<?php
		$result = $mysql->query("SELECT `name`, `emblem_data`, `guild_lv`, `exp`, `guild_id`, `average_lv` FROM `guild` WHERE `guild_id` > '1' ORDER BY `guild_lv` DESC, `exp` DESC LIMIT 25", $CONFIG['DBMain']);

		for( $i = 1; ($line = $mysql->fetcharray($result)); $i++ )
		{
			$emblems[$line['guild_id']] = $line['emblem_data'];

			echo '
				<tr>
					<td align="left">' . $i . '</td>
					<td align="center"><img src="emblema.php?data=' . $line['guild_id'] . '" alt="' . htmlformat($line['name']) . '"></td>
					<td align="left">
						<span title="Ver Profile de Guild" style="color:#000000; cursor:pointer" onMouseOver="this.style.color=#0000FF" onMouseOut="this.style.color=#000000" onClick="return LINK_ajax(\'guildprofile.php?guild_id=' . $line['guild_id'] . '\',\'main_div\');">
							' . htmlformat($line['name']) . '
						</span>
					</td>
					<td align="left">' . $line['guild_lv'] . '</td>
					<td align="right">' . moneyformat($line['exp']) . '</td>
					<td align="right">' . $line['average_lv'] . '</td>
				</tr>
			';
		}
	}
	else if( $_GET['opt'] == 1 )
	{ // WoE Castles
		$castles = $_SESSION[$CONFIG['Name'] . 'castles'];
		$oclastes = array(
			0 => "aldeg_cas01",
			1 => "aldeg_cas02",
			2 => "aldeg_cas03",
			3 => "aldeg_cas04",
			4 => "aldeg_cas05",
			5 => "gefg_cas01",
			6 => "gefg_cas02",
			7 => "gefg_cas03",
			8 => "gefg_cas04",
			9 => "gefg_cas05",
			10 => "payg_cas01",
			11 => "payg_cas02",
			12 => "payg_cas03",
			13 => "payg_cas04",
			14 => "payg_cas05",
			15 => "prtg_cas01",
			16 => "prtg_cas02",
			17 => "prtg_cas03",
			18 => "prtg_cas04",
			19 => "prtg_cas05",
			20 => "nguild_alde",
			21 => "nguild_gef",
			22 => "nguild_pay",
			23 => "nguild_prt",
			24 => "schg_cas01",
			25 => "schg_cas02",
			26 => "schg_cas03",
			27 => "schg_cas04",
			28 => "schg_cas05",
			29 => "arug_cas01",
			30 => "arug_cas02",
			31 => "arug_cas03",
			32 => "arug_cas04",
			33 => "arug_cas05"
		);

?>
	<table width="600">
		<tr>
			<td align="left" class="head">#</td>
			<td align="center" class="head">Emblema</td>
			<td align="left" class="head">Nombre</td>
			<td align="center" class="head">Castillo</td>
			<td align="center" class="head">ECO</td>
			<td align="center" class="head">DEF</td>
		</tr>
<?php
		$result = $mysql->query("
			SELECT
				`guild_castle`.`castle_id`, `guild_castle`.`economy`, `guild_castle`.`defense`,
				`guild`.`guild_id`, `guild`.`name`, `guild`.`emblem_data`
			FROM
				`guild_castle` LEFT JOIN `guild` ON `guild`.`guild_id` = `guild_castle`.`guild_id`
			WHERE
				`guild_castle`.`guild_id` > '0'
			ORDER BY
				`guild_castle`.`castle_id`
			", $CONFIG['DBMain']);

		for( $i = 1; ($line = $mysql->fetcharray($result)); $i++ )
		{
			if( isset($castles[$line['castle_id']]) )
				$cname = $castles[$line['castle_id']];
			else
				continue;

			$emblems[$line['guild_id']] = $line['emblem_data'];

			echo '
				<tr>
					<td align="left">' . $i . '</td>
					<td align="center"><img src="emblema.php?data=' . $line['guild_id'] . '" alt="' . htmlformat($line['name']) . '"></td>
					<td align="left">
						<span title="Ver Profile de Guild" style="color:#000000; cursor:pointer" onMouseOver="this.style.color=#0000FF" onMouseOut="this.style.color=#000000" onClick="return LINK_ajax(\'guildprofile.php?guild_id=' . $line['guild_id'] . '\',\'main_div\');">
							' . htmlformat($line['name']) . '
						</span>
					</td>
					<td align="center">' . $cname . '<br>' . $oclastes[$line['castle_id']] . '</td>
					<td align="center">' . $line["economy"] . '</td>
					<td align="center">' . $line["defense"] . '</td>
				</tr>
			';
		}
	}
	else /* if( $_GET['opt'] == 2 ) */
	{
		$regions = array(
			1 => 'Prontera, Capital of Rune Midgard',
			7 => 'Payon Town',
			8 => 'Morroc Town',
			9 => 'Wootan Tribe\'s Village, Umbala',
			10 => 'Beach Town, Comodo',
			11 => 'Niflheim, Realm of the Dead',
			12 => 'Aldebaran',
			13 => 'Geffen',
			2 => 'Yuno, Capital of Schwarzwald Republic',
			3 => 'Hugel, the Quaint Garden Village',
			4 => 'Lighthalzen, the City of Prosperity',
			5 => 'Einbroch, the City of Steel',
			6 => 'Einbech, the Mining Village'
		);

?>
	<table width="600">
		<tr>
			<td align="left" class="head">#</td>
			<td align="center" class="head">Emblema</td>
			<td align="left" class="head">Nombre</td>
			<td align="right" class="head">Conquested Region</td>
		</tr>
<?php
		$result = $mysql->query("
			SELECT
				`guild`.`guild_id`, `guild`.`name`, `guild`.`emblem_data`, `guild`.`guild_lv`,
				`mapreg`.`index`
			FROM
				`guild` LEFT JOIN `mapreg` ON `mapreg`.`value` = `guild`.`guild_id`
			WHERE
				`mapreg`.`varname` = '\$Region'
			ORDER BY
				`mapreg`.`index`
			", $CONFIG['DBMain']);

		for( $i = 1; ($line = $mysql->fetcharray($result)); $i++ )
		{
			$emblems[$line['guild_id']] = $line['emblem_data'];
			echo '
				<tr>
					<td align="left">' . $i . '</td>
					<td align="center"><img src="emblema.php?data=' . $line['guild_id'] . '" alt="' . htmlformat($line['name']) . '"></td>
					<td align="left">
						<span title="Ver Profile de Guild" style="color:#000000; cursor:pointer" onMouseOver="this.style.color=#0000FF" onMouseOut="this.style.color=#000000" onClick="return LINK_ajax(\'guildprofile.php?guild_id=' . $line['guild_id'] . '\',\'main_div\');">
							' . htmlformat($line['name']) . '
						</span>
					</td>
					<td align="right">' . htmlformat($regions[$line['index']]) . '</td>
				</tr>
			';
		}
	}

	if( isset($emblems) )
		$_SESSION[$CONFIG['Name'] . 'emblems'] = $emblems;

	echo "
			</table>
		</div>
	";

	if( isset($begin) )
	{
		echo "
			</div>
		";

		closetable();
	}
?>